.github/workflows/tests.yml

```diff
@@ -0,0 +1,59 @@
+name: CI
+on:
+  push:
+    branches:
+      - develop
+      - main
+      - master
+  pull_request:
+  schedule:
+    - cron: '46 3 20 * *'
```

Comment:
Why would we run it on schedule?

Reply:
I recommend some sort of a schedule on a monthly basis because Operating Systems (and particularly their packages) that once could be installed commonly deteriorate over time – as was the case when I needed to switch from mirror.centos.org -> vault.centos.org for some RPM sources.

Having this on a schedule means we can catch when the test suite breaks from these sorts of issues, rather than it having to be done in the next Pull Request that is the unfortunate victim of a broken test suite.

Reply:
I have the same type of practice on some projects, it can help detect troubles early on enough, to avoid technical debt setting in.

That said one warning on this, which I learned on another project:

> scheduled workflows are disabled automatically after 60 days of repository inactivity.

https://docs.github.com/en/actions/managing-workflow-runs/disabling-and-enabling-a-workflow

This repo will likely be shut down at some point due to low activity!

Reply:
(maybe it is worth adding a little comment with why the schedule, and link to the doc above too)

Reply:
Great point, added a comment in there now.

Reply:
Make sense, thanks for explaining! :)